src/components/v5/common/Fields/CardSelect/CardSelect.tsx

```diff
@@ -170,6 +174,21 @@ function CardSelect<TValue = string>({
                                       onClick: () => {
                                         onChange(optionValue);
                                         toggleSelectOff();
+
+                                        // These types are a mess!
+                                        // React components that are supposed to be strings
+                                        // And strings that are supposed to be something else
+                                        uiEvents.emit(UIEvent.selectTeam, {
+                                          name:
+                                            // @ts-ignore
+                                            label?.props?.children?.[1] ||
+                                            'Unnamed Team',
+                                          // @ts-ignore
+                                          teamId: optionValue.slice(
+                                            // @ts-ignore
+                                            optionValue.indexOf('=') + 1,
+                                          ),
+                                        });
```

Comment:
TODO: Besides the types being bad, this needs to be refactored as this component is used in some other places, making tracking the domain selector provider false positives